File: MD.CMS.BusinessLogic.Core/DataAccess/Controllers/V2/Options/Interfaces/IFolderRequestOptions.cs
using System.Collections.Generic;

namespace MD.CMS.BusinessLogic.Core.DataAccess.Controllers.V2.Options.Interfaces
{
    public interface IFolderRequestOptions : IPageableRequestOptions, ISearchableRequestOptions, ISortableRequestOptions<Enumerations.Mapping.FolderEnum>
    {
        IEnumerable<long> FolderIds { get; }
        IEnumerable<string> Paths { get; }
        bool FillParent { get; }
        bool FillAllParents { get; }
        bool FillContentTypeDefinitions { get; }
        int Depth { get; }
        bool FillContents { get; }
        bool FillChildren { get; }
        bool FillTemplates { get; }
        FolderRequestOptions ChildFolderRequestOptions { get; }
        FolderRequestOptions ParentFolderRequestOptions { get; }
        DataBoundContentRequestOptions ContentRequestOptions { get; }
        long? ParentId { get; }
        bool OnlyPublished { get; }
        int Lcid { get; }
    }
}

File: MD.CMS.Administration/MD.CMS.Administration.Core.Web/scripts/businessLogic/dataAccess/controllers/options/v2/iFolderRequestOptions.ts
/// <reference path="./enums/folderEnum.ts" />

namespace mdBusinessLogic.dataAccess.controllers.options.v2 {
    export interface iFolderRequestOptions extends iSortableRequestOptions<enums.folderEnum>, iPageableRequestOptions, iSearchableRequestOptions {
        FolderIds?: Array<number>;
        Paths?: Array<string>;
        FillParent?: boolean;
        FillAllParents?: boolean;
        FillContentTypeDefinitions?: boolean;
        Depth?: number;
        FillContents?: boolean;
        FillChildren?: boolean;
        ChildFolderRequestOptions?: iFolderRequestOptions;
        ParentFolderRequestOptions?: iFolderRequestOptions;
        ContentRequestOptions?: iContentRequestOptions;
        ParentId?: number;
        OnlyPublished?: boolean;
        Lcid?: number;
        FillTemplates?: boolean;
    }
}
